homeworks/12-mobx-mars/src/store/marsViewerState.ts

```diff
@@ -0,0 +1,30 @@
+import {makeAutoObservable} from 'mobx'
+import {IMarsViewerState, PhotoType, Tab} from '../types/types'
+import {fetchPhotos} from '../utilities/fetchPhotos'
+
+
+export class MarsViewerState implements IMarsViewerState {
+    public activeTab: Tab = Tab.GALLERY
+    public currentSolution: number = 0
+    public photosToShow: Array<PhotoType> = new Array<PhotoType>()
+    public favourites: Map<string, PhotoType> = new Map<string, PhotoType>()
+    public alreadyLoaded: Map<number, Array<PhotoType>> = new Map<number, Array<PhotoType>>()
+    public loaded: boolean = false
+    public loading: boolean = false
+
+    public constructor(defaultSolution: number) {
+        this.currentSolution = defaultSolution
+        makeAutoObservable(this, {}, {deep: true})
+    }
+
+    public loadNewSolution(): void {
+        !this.loading && fetchPhotos(this)
```

Comment:
This is not a clear practic pass model to the utility function. The function `fetch` knows a lot about the details of the implementation of the `MarsViewerState` class. Now, if you change the details of your model, you will need to change the implementation of the utility function.